src/solvers/smt/smt_conv.cpp

```diff
@@ -2595,6 +2620,45 @@ expr2tc smt_convt::get_by_ast(const type2tc &type, smt_astt a)
   }
 }
 
+double smt_convt::convert_rational_to_double(
+  const BigInt &numerator,
+  const BigInt &denominator)
+{
+  constexpr size_t BUFFER_SIZE = 1024;
+
+  std::vector<char> num_buffer(BUFFER_SIZE, '\0');
+  std::vector<char> den_buffer(BUFFER_SIZE, '\0');
+
+  numerator.as_string(num_buffer.data(), BUFFER_SIZE, 10);
+  denominator.as_string(den_buffer.data(), BUFFER_SIZE, 10);
+
+  // Force null termination to prevent over-read
+  num_buffer[BUFFER_SIZE - 1] = '\0';
+  den_buffer[BUFFER_SIZE - 1] = '\0';
+
+  // Use bounded string length check
+  size_t num_len = strnlen(num_buffer.data(), BUFFER_SIZE);
+  size_t den_len = strnlen(den_buffer.data(), BUFFER_SIZE);
+
+  if (num_len >= BUFFER_SIZE - 1 || den_len >= BUFFER_SIZE - 1)
+  {
+    log_warning(
+      "BigInt to string conversion may have been truncated - buffer too small");
+    return 0.0;
+  }
+
+  try
+  {
+    double num_val = std::stod(num_buffer.data());
+    double den_val = std::stod(den_buffer.data());
+    return (den_val != 0.0) ? num_val / den_val : 0.0;
+  }
+  catch (const std::exception &)
+  {
+    return 0.0;
```

Comment:
When do we expect an exception to be thrown here?
Shouldn't we log a warning in that case?

Reply:
Thanks, @intrigus-lgtm.

Can I ask you to submit a PR with your suggestion?